cmd/queue/main_test.go

```diff
@@ -443,6 +460,58 @@ func TestQueueTraceSpans(t *testing.T) {
 	}
 }
 
+func BenchmarkProxyHandler(b *testing.B) {
+	params := queue.BreakerParams{QueueDepth: 10, MaxConcurrency: 10, InitialCapacity: 10}
+	breaker := queue.NewBreaker(params)
+	// Make the channel as big as possible to account for the largest b.N
+	reqChan := make(chan queue.ReqEvent, 10000000)
+	h := proxyHandler(reqChan, breaker, true /*tracingEnabled*/, fakeHandler{})
+	req := httptest.NewRequest(http.MethodPost, "http://example.com", nil)
+	req.Header.Set(network.OriginalHostHeader, wantHost)
+	resp := httptest.NewRecorder()
+
+	b.Run(fmt.Sprint("sequential"), func(b *testing.B) {
+		for j := 0; j < b.N; j++ {
+			h(resp, req)
+		}
+	})
+
+	b.Run(fmt.Sprint("parallel"), func(b *testing.B) {
+		b.RunParallel(func(pb *testing.PB) {
+			respParallel := httptest.NewRecorder()
+			for pb.Next() {
+				h(respParallel, req)
+			}
+		})
+	})
+}
+
+func BenchmarkProxyHandlerInfiniteBreaker(b *testing.B) {
+	params := queue.BreakerParams{QueueDepth: 10000000, MaxConcurrency: 10000000, InitialCapacity: 10000000}
```

Comment:
@vagababov @markusthoemmes I'm not sure what's the best way to make an infinite breaker here. Do you have any suggestions?

Reply:
I said that wrong: I think the queue-proxy just passes a `nil` in `containerConcurrency==0`. Let's do that.